### PR TITLE
Add java spring middleware

### PR DESCRIPTION
**Pull request by BogdanIrimie**
closes https://github.com/aserto-dev/workspace/issues/255

### CODE REVIEW

**Comment by gimmyxd** [comment]
@BogdanIrimie i think we need to add `java` or `spring` to https://github.com/aserto-dev/topaz-website/blob/d648488891cc26cd7abefce0bc4ecc645ca1a956/docusaurus.config.js#L157 in order to have syntax highlight

File: docs/software-development-kits/java/middleware.mdx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+* converts uppercase characters in the URL path to lowercase
+
+As with the identity contest, you can implement your own PolicyPathMapper by implementing the `PolicyPathMapper`
+interface and creating a bean that returns it.
+
+E.g.
+```java
+    @Bean
+    public PolicyMapper policyMapperDiscoverer() {
+        return new MyCustomPolicyPathMapper(handlerMapping);
```
Comment (ronenh) [inline docs/software-development-kits/java/middleware.mdx:94]: What is `handlerMapping`? It's not defined in this snippet/

File: docs/software-development-kits/java/middleware.mdx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    }
+```
+
+## Resource
+A resource can be any structured data that the authorization policy uses to evaluate decisions.
+By default, the middleware does not include a resource in authorization calls.
```
Comment (ronenh) [inline docs/software-development-kits/java/middleware.mdx:100]: I think the default behavior should be that all path parameters are sent in the resource context.
For example, if the route being called is `/api/users/{user_id}` then, by default, the middleware should extract the `user_id` parameter and send it in the resource context.
In other words a `GET /api/users/1234` request should result in the following resource context:
```json
{
  "user_id": 1234
}
```

File: docs/software-development-kits/java/middleware.mdx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+In order for Spring to discovery the filter, we need to add the `@ComponentScan` annotation to the main class of the application.
+```java
+@ComponentScan("com.aserto")
+```
+
+## Configuration
+
+The authorization filter accepts the fallowing configuration parameters:
+
+| Parameter name | Default Value | Description |
```
Comment (ronenh) [inline docs/software-development-kits/java/middleware.mdx:41]: I would separate the config elements that are only relevant in the hosted environment. Perhaps split the table in two and add:

```
The following configuration settings are available when connecting the middleware to Aserto's hosted authorizer service:
| Parameter name | Default Value | Description |
aserto.authorizer.apiKey | N/A | API key to use authenticate with the hosted authorizer service |
| aserto.tenantId | N/A | The tenant ID to use when connecting to the hosted authorizer service |
| aserto.authorizer.policyName | N/A | The name of the policy in which decisions are evaluated |
| aserto.authorizer.policyLabel | N/A | The instance label of the policy in which decisions are evaluated |
```

File: docs/software-development-kits/java/middleware.mdx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+| aserto.authorization.host | localhost | The hostname of the authorizer service |
+| aserto.authorization.port | 8282 | The port of the authorizer service |
+| aserto.authorizer.insecure | false | Use insecure connection to the authorizer service |
+| aserto.authorizer.cert.path | N/A | The path to the authorizer service certificate, usefull when using local topaz with self signed certificates |
+| aserto.authorizer.apiKey | N/A | The API key to use when connecting to the authorizer service |
+| aserto.tenantId | N/A | The tenant ID to use when connecting to the authorizer service |
+| aserto.authorizer.policyName | N/A | The policy name used by the authorizer |
+| aserto.authorizer.policyLabel | N/A | The policy label used by the authorizer |
+| aserto.authorizer.decision | N/A | The decision used by the authorizer |
+| logging.level.com.aserto | N/A | The log level for the Aserto Middleware |
```
Comment (ronenh) [inline docs/software-development-kits/java/middleware.mdx:53]: Is there really no default value for the log level? Does that mean logging is disabled by default?

**Comment by BogdanIrimie** [comment]
> highlight

That's a great idea, added `java` as a language
https://prismjs.com/#supported-languages

File: docs/software-development-kits/java/middleware.mdx (hunk 1 above)
Comment (BogdanIrimie) [reply docs/software-development-kits/java/middleware.mdx:94]: This is a leftover from the `HttpPathPolicyMapper`. Removed the argument from the constructor as the number and type of arguments will be decided by the one that writes the `MyCustomPolicyPathMapper ` constructor.

File: docs/software-development-kits/java/middleware.mdx (hunk 2 above)
Comment (ogazitt) [reply docs/software-development-kits/java/middleware.mdx:100]: that is consistent with how the rest of our middlewares work

Comment (BogdanIrimie) [reply docs/software-development-kits/java/middleware.mdx:100]: Added the `PathParamsResourceMapper` to the [aserto-spring](https://github.com/aserto-dev/aserto-spring/commit/73739d8660f2ab5d93dc0d3f237e43026cf52276) and make it the default mapper.
I updated the docs to reflect this.

File: docs/software-development-kits/java/middleware.mdx (hunk 4 above)
Comment (BogdanIrimie) [reply docs/software-development-kits/java/middleware.mdx:53]: This is a mistake, the default is INFO

File: docs/software-development-kits/java/middleware.mdx (hunk 3 above)
Comment (BogdanIrimie) [reply docs/software-development-kits/java/middleware.mdx:41]: Splitted the config as suggested

**Review by ogazitt — APPROVED** [review]

**Review by ronenh — APPROVED** [review]

**Pull request merged**